### PR TITLE
HTCONDOR-1000  Miscellaneous pre-release improvements.

### PR DESCRIPTION
**Pull request by Todd-L-Miller**
This is probably one of those PRs which will probably only make sense if reviewed one commit at a time.

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed
- [ ] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_tools/htcondor_cli/annex_create.py [hunk 1]
```diff
@@ -668,53 +657,34 @@ def annex_create(
...
-        control_path,
     )
 
+    if rc == 0:
+        logger.info(f"... remote SLURM job submitted.")
+    else:
+        error = f"Failed to start annex, SLURM returned code {rc}"
+        try:
+            schedd.act(htcondor.JobAction.Remove, f'ClusterID == {cluster_id}', error)
+        except:
```
Comment (jasoncpatton) [inline src/condor_tools/htcondor_cli/annex_create.py:687]:
```suggestion
        except Exception:
```

Never use bare `except:` (see e.g. https://stackoverflow.com/a/54948581 )

Comment (jasoncpatton) [reply src/condor_tools/htcondor_cli/annex_create.py:687]: And I'm noticing now that I also failed to fix this earlier in `annex_create.py`:

```python
    except:
        raise RuntimeError(f"Failed to submit state-tracking job, aborting.")
```

**Review by jasoncpatton — CHANGES_REQUESTED** [review]

File: src/condor_tools/htcondor_cli/annex_create.py (hunk 1 above)
Comment (Todd-L-Miller) [reply src/condor_tools/htcondor_cli/annex_create.py]: Done.

**Review by jasoncpatton — APPROVED** [review]
LGTM, docs/test waiting for HTCONDOR-958

**Pull request merged**